stacker/stack.py

```diff
@@ -192,3 +193,94 @@ def resolve(self, context, provider):
 
     def set_outputs(self, outputs):
         self.outputs = outputs
+
+    def should_submit(self):
+        """Tests whether this stack should be submitted to CF
+
+        Returns:
+            bool: If the stack should be submitted, return True.
+
+        """
+
+        if self.enabled:
+            return True
+
+        logger.debug("Stack %s is not enabled.  Skipping.", self.name)
+        return False
+
+    def should_update(self):
+        """Tests whether this stack should be submitted for updates to CF.
+
+        Returns:
+            bool: If the stack should be updated, return True.
+
+        """
+
+        if self.locked:
+            if not self.force:
+                logger.debug("Stack %s locked and not in --force list. "
+                             "Refusing to update.", self.name)
+                return False
+            else:
+                logger.debug("Stack %s locked, but is in --force "
+                             "list.", self.name)
+        return True
+
+
+class ExternalStack(Stack):
+    """Represents gathered information about an existing external stack
+
+    Args:
+        definition (:class:`stacker.config.ExternalStack`): A stack definition.
+        context (:class:`stacker.context.Context`): Current context for
+            building the stack.
+
+    """
+
+    def __init__(self, definition, context):
+        self.name = definition.name
+        stack_name = definition.stack_name or self.name
+        self.fqn = definition.fqn or context.get_fqn(stack_name)
+        self.region = definition.region
+        self.profile = definition.profile
+        self.definition = definition
+        self.context = context
+        self.outputs = None
+
+    @property
+    def requires(self):
+        return set()
+
+    @property
+    def stack_policy(self):
+        return None
+
+    @property
+    def blueprint(self):
+        return None
+
+    @property
+    def tags(self):
+        return dict()
+
+    @property
+    def parameter_values(self):
+        return dict()
+
+    @property
+    def required_parameter_definitions(self):
+        return dict()
+
+    def resolve(self, context, provider):
+        pass
+
+    def set_outputs(self, outputs):
+        self.outputs = outputs
+
+    def should_submit(self):
```

Comment:
What does this end up looking like in the CLI output when we hit an external stack?  Does it say it's submitting/creating?

Reply:
I believe it prints "skipped". Do you think we should add an additional status to print something different?

Reply:
Hmm - yeah, that might be useful.  It can still be skipped, just with a slightly different message maybe?